pkg/apis/gcp/cache/v1alpha1/cloudmemorystore_instance_types.go

```diff
@@ -208,27 +210,67 @@ type CloudMemorystoreInstanceList struct {
 
 // NewCloudMemorystoreInstanceSpec creates a new CloudMemorystoreInstanceSpec
 // from the given properties map.
-func NewCloudMemorystoreInstanceSpec(properties map[string]string) *CloudMemorystoreInstanceSpec {
+func NewCloudMemorystoreInstanceSpec(st CloudMemorystoreInstanceClassSpecTemplate) *CloudMemorystoreInstanceSpec {
 	spec := &CloudMemorystoreInstanceSpec{
 		ResourceSpec: corev1alpha1.ResourceSpec{
 			ReclaimPolicy: corev1alpha1.ReclaimRetain,
 		},
 
+		CloudMemorystoreInstanceParameters: st.CloudMemorystoreInstanceParameters,
 		// Note that these keys should match the JSON tags of their respective
 		// CloudMemorystoreInstanceSpec fields.
-		Region:                properties["region"],
-		Tier:                  properties["tier"],
-		LocationID:            properties["locationId"],
-		AlternativeLocationID: properties["alternativeLocationId"],
-		ReservedIPRange:       properties["reservedIpRange"],
-		AuthorizedNetwork:     properties["authorizedNetwork"],
-		RedisVersion:          properties["redisVersion"],
-		RedisConfigs:          util.ParseMap(properties["redisConfigs"]),
-	}
+		// Region:                properties["region"],
```

Comment:
Commented out but not removed yet to show the difference in parsing a properties map vs just setting the parameters to the parameters struct defined in the class. Some modification / defaulting is done in the Cloud Memorystore controller, and it would probably be useful to standardize on one of the following patterns:
- Do all class properties validation / defaulting / etc. in functions like the `NewCloudMemorystoreInstanceSpec()`
- Reserve functions like `NewCloudMemorystoreInstanceSpec()` for only setting the managed resource parameters equal to the parameters defined on the class

We should also note that defaulting is now enabled of CRD's: https://kubernetes.io/docs/tasks/access-kubernetes-api/custom-resources/custom-resource-definitions/#defaulting

Reply:
Absolutely. Kind of building on my response to your comment above, I didn't know if there was real benefit to setting values in this function as opposed to the `Configurator`. My vote would be to get rid of this function because it makes sense for all logic to live in the function that is meant to "configure" the resource.